Lib/typing.py

```diff
@@ -440,8 +442,44 @@ def NoReturn(self, parameters):
       def stop() -> NoReturn:
           raise Exception('no way')
 
-    This type is invalid in other positions, e.g., ``List[NoReturn]``
-    will fail in static type checkers.
+    NoReturn can also be used as a bottom type, a type that
+    has no values. Starting in Python 3.11, the Never type should
+    be used for this concept instead. Type checkers should treat the two
+    equivalently.
+
+    """
+    raise TypeError(f"{self} is not subscriptable")
+
+# This is semantically identical to NoReturn, but it is implemented
+# separately so that type checkers can distinguish between the two
+# if they want.
+@_SpecialForm
+def Never(self, parameters):
+    """The bottom type, a type that has no members.
+
+    The `bottom type <https://en.wikipedia.org/wiki/Bottom_type>`_,
+    a type that has no members.
```

Comment:
There's repetition here on lines 458-461, and using Rest syntax in a docstring is a little weird

Reply:
Good catch, will remove